Taarafo.Core/Controllers/CommentsController.cs

```diff
@@ -54,5 +56,28 @@ public async ValueTask<ActionResult<Comment>> PostCommentAsync(Comment comment)
                 return InternalServerError(commentServiceException);
             }
         }
+
+        [HttpGet]
+        public ActionResult<IQueryable<Comment>> GetAllComments()
+        {
+            try
+            {
+                IQueryable<Comment> retrievedComments =
+                    this.commentService.RetrieveAllComments();
+
+                return Ok(retrievedComments);
+            }
+            catch (CommentDependencyException commentDependencyException)
+            {
+                return InternalServerError(commentDependencyException);
+            }
+            catch (CommentServiceException commentServiceException)
+            {
+                return InternalServerError(commentServiceException);
+            }
+        }
+
+        private static string GetInnerMessage(Exception exception) =>
+            exception.InnerException.Message;
```

Comment:
Why do you need this?

Reply:
@hassanhabib removed it.  Should have been removed with the exception fix.